### PR TITLE
feat: add confirmation modal for delete actions [WD-13690]

### PR DESCRIPTION
**Pull request by mas-who**
## Done
 - introduced confirmation modal for revoke cluster token and delete pending cluster actions.

## QA
 - revoke a token. Make sure the confirmation modal displays and has reasonable wording.
 - delete a pending cluster. Make sure the confirmation modal displays and has reasonable wording.

## Screenshots
Screenshot from 2024-08-01 13-09-36
Screenshot from 2024-08-01 13-10-14

### CODE REVIEW

File: ui/src/pages/clusters/RevokeTokenButton.tsx [hunk 1]
```diff
@@ -25,17 +27,34 @@ const RevokeTokenButton: FC<Props> = ({ token }) => {
...
+    <ConfirmationButton
       appearance="negative"
-      onClick={() => void handleDeleteToken()}
+      loading={loading}
+      className="u-no-margin--bottom"
+      confirmationModalProps={{
+        title: "Confirm revoke",
+        children: (
+          <p>
+            This will permanently delete the token for cluster{" "}
```
Comment (Kxiru) [inline ui/src/pages/clusters/RevokeTokenButton.tsx]: "delete" => "revoke"?

**Review by Kxiru — DISMISSED** [review]
Demo looks good too!
Worth putting a screenshot in the PR details?
LGMT :)

File: ui/src/pages/clusters/RevokeTokenButton.tsx (hunk 1 above)
Comment (mas-who) [reply ui/src/pages/clusters/RevokeTokenButton.tsx]: good catch :+1:

**Comment by mas-who** [comment]
> Demo looks good too! Worth putting a screenshot in the PR details? LGMT :)

I've added the screenshots :+1:

**Review by Kxiru — APPROVED** [review]
Nice job, Mason :)

**Pull request merged**